File: packages/schemas/src/telegram.ts
import { z } from 'zod';
import { Iso8601Schema } from './common';

/** Telegram 알림 대상 — `docs/integrations/telegram.md` §6 (T-106).
 * bot token은 받지 않는다(§1) — 시스템 봇을 chat에 추가한 뒤 chat_id만 등록.
 */
export const TelegramTargetCreateSchema = z.object({
  telegram_chat_id: z.string().min(1).max(64),
  telegram_label: z.string().max(80).nullable().optional(),
  telegram_message_thread_id: z.string().max(64).nullable().optional(),
  is_default: z.boolean().optional().default(false),
});
export type TelegramTargetCreate = z.infer<typeof TelegramTargetCreateSchema>;

export const TelegramTargetSchema = z.object({
  id: z.string().uuid(),
  telegram_chat_id: z.string(),
  telegram_chat_type: z.string().nullable(),
  telegram_message_thread_id: z.string().nullable(),
  telegram_label: z.string().nullable(),
  title_snapshot: z.string().nullable(),
  is_default: z.boolean(),
  is_enabled: z.boolean(),
  last_verified_at: Iso8601Schema.nullable(),
  last_send_status: z.string().nullable(),
  created_at: Iso8601Schema,
});
export type TelegramTarget = z.infer<typeof TelegramTargetSchema>;

File: apps/api/app/schemas/telegram.py
"""Telegram 알림 대상 schemas — `docs/integrations/telegram.md` §6. T-106."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field, field_validator


class TripTelegramTargetLink(BaseModel):
    """`POST /trips/{trip_id}/telegram-targets` 본문 (§6.5)."""

    telegram_target_id: uuid.UUID


class TelegramTargetCreate(BaseModel):
    """`POST /users/me/telegram-targets` 본문.

    bot token은 받지 않는다(§1) — Pinvi 시스템 봇을 사용자가 자기 chat에 추가한 뒤
    chat_id만 등록한다. `telegram_bot_token_ref`는 현 단계 `system`만 허용.
    """

    telegram_chat_id: str = Field(min_length=1, max_length=64)
    telegram_label: str | None = Field(default=None, max_length=80)
    telegram_message_thread_id: str | None = Field(default=None, max_length=64)
    is_default: bool = False

    @field_validator("telegram_chat_id")
    @classmethod
    def _strip_chat_id(cls, value: str) -> str:
        stripped = value.strip()
        if not stripped:
            raise ValueError("telegram_chat_id는 비어 있을 수 없습니다.")
        return stripped

    @field_validator("telegram_label")
    @classmethod
    def _normalize_label(cls, value: str | None) -> str | None:
        if value is None:
            return None
        stripped = value.strip()
        return stripped or None


class TelegramTargetResponse(BaseModel):
    id: uuid.UUID
    telegram_chat_id: str
    telegram_chat_type: str | None
    telegram_message_thread_id: str | None
    telegram_label: str | None
    title_snapshot: str | None
    is_default: bool
    is_enabled: bool
    last_verified_at: datetime | None
    last_send_status: str | None
    created_at: datetime
